### DISCUSSION TITLE
Set IO striping

### DISCUSSION THREAD

**Discussion by jacwah** [ExtremeFLOW/neko#389] [Ideas]
Nek users on Lustre usually have to set the file striping on each output directory to get good IO performance, i.e. command `lfs setstripe -c 1 .`. We could make this more convenient for users by doing it automatically on a per-file basis where relevant. MPICH has builtin support for this using `striping_factor` key in an `MPI_Info` object. (Note that the `MPICH_MPIIO_HINTS` environment variable can still override these settings if necessary.)

Ref: https://www.alcf.anl.gov/files/CompPerfWorkshop_May2019_IOopt_harms.pdf

**Comment by steven-chien** [comment]
There could be a portability issue. For example, OMPI (that does not use ROMIO) has different hints on that: https://www.open-mpi.org/faq/?category=ompio#fs-parametesrs
A possible way is to make the code path a compile-time option with `configure`?

**Reply by jacwah** [reply]
Good point. It seems like the corresponding option for OMPI would be `stripe_size`. The simplest solution would be to just set both, it probably would not hurt to specify an Info option that is not used. For a more elegant solution, one could try to detect the MPI using some m4 magic as well.

**Comment by jacwah** [comment]
Discussed this with Philipp Schlatter in RT ticket 247964.
